src/librustc/ty/print/pretty.rs

```diff
@@ -969,69 +987,131 @@ pub trait PrettyPrinter<'tcx>:
-                }
+    /// This is overridden for MIR printing because we only want to hide alloc ids from users, not
+    /// from MIR where it is actually useful.
+    fn pretty_print_const_pointer(
+        mut self,
+        _: Pointer,
+    ) -> Result<Self::Const, Self::Error> {
+        define_scoped_cx!(self);
+        p!(write("{{pointer}}"));
+        Ok(self)
+    }
+
+    fn pretty_print_byte_str(
+        mut self,
+        byte_str: &'tcx [u8],
+    ) -> Result<Self::Const, Self::Error> {
+        define_scoped_cx!(self);
+        p!(write("b\""));
+        for &c in byte_str {
+            for e in std::ascii::escape_default(c) {
+                self.write_char(e as char)?;
             }
-        };
+        }
+        p!(write("\""));
         Ok(self)
     }
+
+    fn pretty_print_const_value(
+        mut self,
+        ct: ConstValue<'tcx>,
+        ty: Ty<'tcx>,
+    ) -> Result<Self::Const, Self::Error> {
+        define_scoped_cx!(self);
+
+        if self.tcx().sess.verbose() {
+            p!(write("ConstValue({:?}: {:?})", ct, ty));
+            return Ok(self);
+        }
+
+        let u8 = self.tcx().types.u8;
+
+        match (ct, &ty.kind) {
+            (ConstValue::Scalar(scalar), _) => self.pretty_print_const_scalar(scalar, ty),
+            (
+                ConstValue::Slice { data, start, end },
+                ty::Ref(_, ty::TyS { kind: ty::Slice(t), .. }, _),
+            ) if *t == u8 => {
+                // The `inspect` here is okay since we checked the bounds, and there are
+                // no relocations (we have an active slice reference here). We don't use
+                // this result to affect interpreter execution.
+                let byte_str = data.inspect_with_undef_and_ptr_outside_interpreter(start..end);
+                self.pretty_print_byte_str(byte_str)
+            },
+            (
+                ConstValue::Slice { data, start, end },
+                ty::Ref(_, ty::TyS { kind: ty::Str, .. }, _),
+            ) => {
+                // The `inspect` here is okay since we checked the bounds, and there are no
+                // relocations (we have an active `str` reference here). We don't use this
+                // result to affect interpreter execution.
+                let slice = data.inspect_with_undef_and_ptr_outside_interpreter(start..end);
+                let s = ::std::str::from_utf8(slice)
+                    .expect("non utf8 str from miri");
+                p!(write("{:?}", s));
+                Ok(self)
+            },
+            (ConstValue::ByRef { alloc, offset }, ty::Array(t, n)) if *t == u8 => {
+                let n = n.eval_usize(self.tcx(), ty::ParamEnv::empty());
+                let n = Size::from_bytes(n);
+                let ptr = Pointer::new(AllocId(0), offset);
+
+                let byte_str = alloc.get_bytes(&self.tcx(), ptr, n).unwrap();
+                p!(pretty_print_byte_str(byte_str));
```

Comment:
Shouldn't this print something like `*b"..."`, i.e. a dereference of a byte string literal?

Reply:
Actually, do you even need this special-case? This is just an array of bytes, when does this even get used right now?

Reply:
Idk, I added it because it was easy to do and didn't see why we should not print it

Reply:
Well, it's technically wrong because this array is by-value, but `b"..."` is a reference :P.

Reply:
I'm printing `*b"..."` now